lib/rubocop/cop/mixin/minitest_exploration_helpers.rb

```diff
@@ -66,7 +66,18 @@ def class_def_nodes(class_node)
 
       # support https://api.rubyonrails.org/classes/ActiveSupport/Testing/Declarative.html
       def test_method_calls(class_node)
-        class_node.each_child_node(:block).select { |blk| blk.each_child_node(:send).first&.method_name.to_s == 'test' }
+        all_blocks(class_node).select { |blk| blk.each_child_node(:send).first&.method_name.to_s == 'test' }
+      end
+
+      def all_blocks(parent_node)
+        parent_node.each_child_node.each_with_object([]) do |node, list|
+          list << node if node.block_type?
+          node.each_child_node.each do |n|
+            list << n if n.block_type?
+            list.concat(all_blocks(n))
+          end
+          list
+        end
       end
```

Comment:
Thank you for quick fixing. Can you update to the code?

```diff
       # support https://api.rubyonrails.org/classes/ActiveSupport/Testing/Declarative.html
       def test_method_calls(class_node)
-        all_blocks(class_node).select { |blk| blk.each_child_node(:send).first&.method_name.to_s == 'test' }
+        block_nodes = class_node.each_descendant(:block)

+        block_nodes.select { |block_node| block_node.method?(:test) }
       end

-      def all_blocks(parent_node)
-        parent_node.each_child_node.each_with_object([]) do |node, list|
-          list << node if node.block_type?
-          node.each_child_node.each do |n|
-            list << n if n.block_type?
-            list.concat(all_blocks(n))
-          end
-          list
-        end
-      end
```

Reply:
That's awesome, thank you!